client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -313,13 +342,13 @@ function createRenewalItemToAddToCart( productAlias, productId, purchaseId, sele
 		{
 			meta,
 			product_slug: productSlug,
-			product_id: productId,
+			product_id: parseInt( String( productId ), 10 ),
 		},
 		{
 			id: purchaseId,
 			domain: selectedSiteSlug,
 		}
-	);
+	) as RequestCartProduct;
```

Comment:
This is fine but I like to use validators/type guards rather than casting to ensure things are as I expect.

Reply:
How would I do that here?

Reply:
Could use a similar approach as the validators we used earlier. Or a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html). You don't have to do an exhaustive check so you may just want to validate certain properties exist? I don't know if it would work but you *could* accept the `CartItemValue` as the input type and just validate the differences between the two types.

Reply:
👍 Done.